src/aiida/tools/dumping/processes.py

```diff
@@ -285,6 +289,53 @@ def _dump_calculation(
                 link_triples=output_links,
             )
 
+        # Additional remote file retrieval should only apply for CalcJobNodes, not CalcFunctionNodes
+        if self.also_remote and isinstance(calculation_node, CalcJobNode):
+            self._dump_calculation_remote_files(calcjob_node=calculation_node, output_path=output_path)
+
+    def _dump_calculation_remote_files(self, calcjob_node: CalcJobNode, output_path: Path) -> None:
+        """Dump the additional remote files attached to a `CalcJobNode` to a specified output path.
+
+        :param calcjob_node: The `CalcJobNode` to be dumped.
+        :param output_path: The path where the files will be dumped.
+        """
+
+        remote_workdir = calcjob_node.get_remote_workdir()
+        if remote_workdir is None:
+            raise NotExistentAttributeError(f"CalcJobNode <{calcjob_node.pk}> doesn't have a `remote_workdir`.")
```

Comment:
This is not caught, do you really want to let this let the entire dump fail? Perhaps just log a warning and move on